xonsh/environ.py

```diff
@@ -120,6 +121,7 @@ def to_debug(x):
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
     'XONSH_HISTORY_SIZE': (is_history_tuple, to_history_tuple, history_tuple_to_str),
     'XONSH_LOGIN': (is_bool, to_bool, bool_to_str),
+    'XONSH_SHOW_TRACEBACK': (is_tb_opt, to_tb_tuple, tb_tuple_to_str),
```

Comment:
I think the tuple idea is a reasonable one.  This is basically what we do for history.  With out something like this we would need to have two envvars, which could be annoying too.  I think that is the real trade off here.